File: Java/Problems/Arrays/KDiffpairs.java
package Problems.Arrays;

/**
 * k-diff pairs
 * Medium
 * 50
 * Given a sorted array A of size n and a number k,
 * return the number of unique pairs which have a difference of k.
 *
 * |arr[i] - arr[j]| = k where i < j
 */
public class KDiffpairs {

    int kDiffPairs (int[] A, int k) {
        // add your logic here
        int count=0;
        int x = 0;
        for (int i=0; i< A.length; i++){
            if (x!=A[i]) {
                x = A[i];
                for (int j = i + 1; j < A.length; j++) {
                    if (Math.abs(x - A[j]) == k) {
                        count++;
                        break;
                    }
                }
            }
        }

        return count;
    }

    public static void main(String[] args) {
        int[] A = new int[]{1, 3, 5, 7, 10};
        KDiffpairs kDiffpairs = new KDiffpairs();

        System.out.println(kDiffpairs.kDiffPairs(A, 2));
        System.out.println(kDiffpairs.kDiffPairs(A, 3));
        System.out.println(kDiffpairs.kDiffPairs(A, 1));
    }
}

File: GO-Lang/Problems/Arrays/KDiffpairs/KDiffpairs.go
package main

import (
	"fmt"
	"math"
)

/**
 * k-diff pairs
 * Medium
 * 50
 * Given a sorted array A of size n and a number k,
 * return the number of unique pairs which have a difference of k.
 *
 * |arr[i] - arr[j]| = k where i < j
 */

func kDiffPairs(A *[]int, k int) int {
	count := 0
	x := 0

	for i := 0; i < len(*A); i++ {
		if (*A)[i] != x {
			x = (*A)[i]
			for j := i; j < len(*A); j++ {
				if int(math.Abs(float64((*A)[j]-x))) == k {
					count++
				}
			}
		}
	}

	return count
}

func main() {
	A := []int{1, 3, 5, 7, 10}
	fmt.Println("-->", kDiffPairs(&A, 2))
	fmt.Println("-->", kDiffPairs(&A, 3))
	fmt.Println("-->", kDiffPairs(&A, 1))
}
